crates/miden-tx/src/executor/mod.rs

```diff
@@ -223,15 +222,15 @@ where
         advice_inputs: AdviceInputs,
         foreign_account_inputs: Vec<AccountInputs>,
     ) -> Result<[Felt; 16], TransactionExecutorError> {
-        let tx_args = TransactionArgs::new(Default::default(), foreign_account_inputs)
+        let mut tx_args = TransactionArgs::new(Default::default(), foreign_account_inputs)
             .with_tx_script(tx_script);
+        tx_args.extend_advice_inputs(advice_inputs);
```

Comment:
Not for this PR, but we could also refactor `TransactionArgs::new` so that it takes `AdviceInputs` instead of `AdviceMap`. That would avoid the need for the extend call.